luaparse.js

```diff
@@ -907,15 +912,27 @@
           // Return it as is, without translating the byte.
           return '\\' + input.slice(sequenceStart, index);
         }
+        return raise({}, errors.hexadecimalDigitExpected, '\\' + input.slice(sequenceStart, index + 2));
+      case '\\':
+      case '\'':
+      case '"':
+      case '[':
+      case ']':
```

Comment:
`"\["` and `"\]"` aren't actually valid escapes. Lua 5.1 treats them as if they were valid, but it's no different from other unrecognised escapes.